plugins/catalog-backend/src/service/router.ts

```diff
@@ -142,7 +142,75 @@ export async function createRouter(
           );
         }
         res.status(200).json(entities[0]);
-      });
+      })
+      .get('/entities/by-uid/:uid/attachments/:key', async (req, res) => {
+        const { uid, key } = req.params;
+        const attachment = await entitiesCatalog.attachment(uid, key, {
+          ifNotMatchEtag: req.header('if-none-match'),
+        });
+
+        if (!attachment) {
+          throw new NotFoundError(
+            `No attachment with key '${key}' found for entity with uid ${uid}'`,
+          );
+        }
+
+        res.header('ETag', attachment.etag);
+
+        if (!attachment.data) {
+          res.status(304).send();
+          return;
+        }
+
+        res
+          .status(200)
+          .contentType(attachment.contentType)
+          .send(attachment.data);
+      })
+      .get(
+        '/entities/by-name/:kind/:namespace/:name/attachments/:key',
+        async (req, res) => {
+          const { kind, namespace, name, key } = req.params;
+          const { entities } = await entitiesCatalog.entities({
+            filter: basicEntityFilter({
+              kind: kind,
+              'metadata.namespace': namespace,
+              'metadata.name': name,
+            }),
+          });
+          if (!entities.length) {
+            throw new NotFoundError(
+              `No entity named '${name}' found, with kind '${kind}' in namespace '${namespace}'`,
```

Comment:
here and below it'd be fine with just `${stringifyEntityRef({ kind, namespace, name })}` for the name imo :)